### PR TITLE
Re-work pre-commit (without the ci variant)

### PR DESCRIPTION
**Pull request by CoMPaTech**
Back to previous - run in worker

## Summary by CodeRabbit

- **New Features**
	- Introduced a `pre-commit` job to enhance code quality checks before preparing the environment.

- **Bug Fixes**
	- Updated Python version in pre-commit configuration for improved compatibility.

- **Documentation**
	- Removed CI status badges from the README for a cleaner presentation.
	- Added a new section in CHANGELOG to note ongoing improvements in the CI process.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,21 +13,61 @@ on:
...
+          restore-keys: |
+            ${{ env.CACHE_VERSION}}-${{ runner.os }}-base-hacore-${{ steps.python.outputs.python-version }}-
+            ${{ env.CACHE_VERSION}}-${{ runner.os }}-base-hacore
+            ${{ env.CACHE_VERSION}}-${{ runner.os }}
+            ${{ env.CACHE_VERSION}}
+      - name: Install pre-commit dependencies
+        if: steps.cache-precommit.outputs.cache-hit != 'true'
+        run: |
+          . venv/bin/activate
+          pre-commit install-hooks
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply .github/workflows/test.yml]: Tnx, adjusted accordingly

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,13 +1,6 @@
...
-ci:
-  # Defer autoupdate to quarterly (there is no 'off' button) to have renovate pick up first
-  autoupdate_schedule: quarterly
-  skip: [test-core-prep, test-pip-prep, testing, quality]
-  submodules: true
-
 default_language_version:
   # force all unspecified python hooks to run python3
-  python: python3.11
+  python: python3.13
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply .pre-commit-config.yaml:3]: At one point we will, not relevant in this PR - also - 313 is considered stable

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@
 name: Test against HA-core (master/released)
 
 env:
-  CACHE_VERSION: 27
+  CACHE_VERSION: 28
   DEFAULT_PYTHON: "3.13"
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply .github/workflows/test.yml]: 3.13 is considered stable :)

**Comment by CoMPaTech** [comment]
Only pre-commit.ci should fail now

**Comment by CoMPaTech** [comment]
Pre-commit.CI turned off for beta and progress (so the USBs and module are still todo)

**Review by bouwew — APPROVED** [review]
LGTM!

**Pull request merged**